### PR TITLE
blend: perform all arithmetic floating pint operations in single precision

### PR DESCRIPTION
**Pull request by rabauke**
This PR is to keep all arithmetic operations, which are performed in a blend operation, in the single precision domain. This givea a (fairly small but easy to get) performance gain. Essentially the double precision functions fabs, fmin and fmax are replaced by the single precision functions fabsf, fminf and fmaxf. Note that M_PI is a double precision constant. Thus, calculations involving M_PI are promoted to double.

### CODE REVIEW

File: src/develop/blend.c [hunk 1]
```diff
@@ -213,9 +213,9 @@ static inline void _Lab_2_LCH(const float *Lab, float *LCH)
   float var_H = atan2f(Lab[2], Lab[1]);
 
   if(var_H > 0.0f)
-    var_H = var_H / (2.0f * M_PI);
+    var_H = var_H / (2.0f * 3.141592654f);
   else
-    var_H = 1.0f - fabs(var_H) / (2.0f * M_PI);
+    var_H = 1.0f + var_H / (2.0f * 3.141592654f);
```
Comment (upegelow) [inline src/develop/blend.c]: What's the rationale for this change?

File: src/develop/blend.c [hunk 2]
```diff
@@ -225,8 +225,8 @@ static inline void _Lab_2_LCH(const float *Lab, float *LCH)
 static inline void _LCH_2_Lab(const float *LCH, float *Lab)
 {
   Lab[0] = LCH[0];
-  Lab[1] = cosf(2.0f * M_PI * LCH[2]) * LCH[1];
-  Lab[2] = sinf(2.0f * M_PI * LCH[2]) * LCH[1];
+  Lab[1] = cosf(2.0f * 3.141592654f * LCH[2]) * LCH[1];
```
Comment (upegelow) [inline src/develop/blend.c]: I suggest to use M_PI_F as a floating point constant at these places.

File: src/develop/blend.c (hunk 1 above)
Comment (rabauke) [reply src/develop/blend.c]: The variable `var_H` is negative in the `else` of `if(var_H > 0.0f)`. Therefore, `1.0f - fabs(var_H)/ (2.0f * M_PI)` is equivalent to `1.0f + var_H / (2.0f * M_PI)` in this particular context.

File: src/develop/blend.c (hunk 2 above)
Comment (rabauke) [reply src/develop/blend.c]: Although, the macro `M_PI_F` is commonly used in the darktable OpenCL sources. It is not defined in any C source file. Note that not even `M_PI` is defined by any <i>standard</i> C header.

Possibly, one should add a new header file to the darktable sources, which defines new macros as `DT_M_PI`, `DT_M_PI_F` etc. for single and double precision representations of common mathematical constants.

Comment (rabauke) [reply src/develop/blend.c]: I replaced the numerical value 3.141592654f by the new macro `DT_M_PI_F`, which has been defined in the new header `common/math.h` and may be useful in other source code files as well.

File: src/develop/blend.c [hunk 3]
```diff
@@ -1113,13 +1114,13 @@ static void _blend_screen(const _blend_buffer_desc_t *bd, const float *a, float
       float lmin = 0.0, lmax, la, lb;
```
Comment (upegelow) [inline src/develop/blend.c]: I suggest to replace the initialization value of lmin to float (here and in the subsequent code parts).

Comment (rabauke) [reply src/develop/blend.c]: This has been fixed too, now.
